src/main/java/com/eppo/sdk/helpers/FetchConfigurationsTask.java

```diff
@@ -15,8 +17,16 @@ public class FetchConfigurationsTask extends TimerTask {
 
   @Override
   public void run() {
-    configurationStore.fetchAndSetExperimentConfiguration();
-    long delay = this.intervalInMillis - ((long) Math.random() * this.jitterInMillis);
+    // Uncaught runtime exceptions will prevent this task from being rescheduled.
+    // As a result, the SDK will continue functioning using the in-memory cache, but will never attempt
+    // to synchronize with Eppo Cloud again.
+    try {
+      configurationStore.fetchAndSetExperimentConfiguration();
+    } catch (Exception e) {
+      log.error("[Eppo SDK] Error fetching experiment configuration", e);
+    }
```

Comment:
**Wouldn't something like `ScheduledExecutorService` be more fault tolerant?**

Yes! `ScheduledExecutorService` has a lot of benefits over `TimerTask`, including fault tolerance.... However, you need to shutdown the executor service when terminating the application. This felt like a heavier change, as we'd need to change the `EppoClient` to either accept a scheduler, or have it's own lifecycle methods to properly shut down the executor. 

Catching/logging an uncaught exceptions felt like a safer, more surgical change.

Reply:
Thanks @simondale00 this is a good pragmatic improvement - we have a backlog item to add more robust remote re-polling inline with our recent improvements to eppo's javascript sdks: https://github.com/Eppo-exp/js-client-sdk-common/commit/c892b5f7a979059cdc6cd87e577bd67f826f5370